src/lib/parallel_scan/scan_state.md

```diff
@@ -0,0 +1,141 @@
+# Scan State Refactoring
+
+The parallel scan state currently is a full-binary tree with leaves (or called `Base` nodes) having values of type `Transaction_with_witness.t` and intermediate nodes (or called `Merge` nodes) having values of type `Ledger_proof_with_sok_message.t`
+
+Everytime a diff is applied, the transactions are transformed to new base jobs and added to the scan state. The diff also includes completed works that correspond to a sequence of jobs that already exist on the scan state. These, when added to scan state, create new merge jobs except when it is for the root node in which case the proof is simply returned as the result.
+
+Currently, parallel scan is implemeted using arrays and involve a lot of index manipulation. This makes the code difficult to read and maintain. This refactoring aims to make the structure declarative and more "functional" given that we can lax a bit on the efficiency since other expensive operations(sparse ledgers for witness) in `apply_diff`.
```

Comment:
implemented

Reply:
there's a word or phrase missing here, maybe this should be "since there are other expensive" ?